### PR TITLE
Modify update method in UsersController

### PR DESCRIPTION
**Pull request by ghost**
-Modified the "update" method for users_controller, since the failure condition is not present at the moment.
-Added rspec for controllers/users_controller.rb
-Added SessionsHelper to rails_helper.rb

### CODE REVIEW

**Comment by swively** [comment]
The title of this pr is misleading because you are also changing behaviour in the user controller - you've removed a flash message for the `update` method's fail path.

Typically tests are added alongside changes or additions, not as standalone pull requests.

File: Gemfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ group :development, :test do
   gem 'rspec-rails', '~> 3.5'
   gem 'simplecov'
   gem 'simplecov-rcov'
+  gem 'pry-rails'
```
Comment (qr8r) [inline Gemfile]: This should probably be its own PR. 

if this commit were on its own I could merge it. Then you could rebase all of your other branches on top of master to get pry in your test environment for all your branches

File: spec/controllers/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      it { is_expected.to be_ok }
+    end
+  end
+
+  describe 'POST #update' do
+    let(:user) do
+      User.create(username: 'Bob', email: 'bob@mail.com', password: '123')
+    end
+    let(:valid_params) { { username: 'Robert' } }
+    let(:invalid_params) { { age: '35' } }
```
Comment (qr8r) [inline spec/controllers/users_controller_spec.rb]: Is this used in this test?

File: Gemfile (hunk 1 above)
Comment (ghost) [reply Gemfile]: Thanks, moved it to its own PR.

**Comment by cbrunsdon** [comment]
jenkins test this please

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  end
+
+  describe 'POST #create' do
+    subject(:post_create) { post :create, params: { user: user_params } }
+    let(:valid_params) do
+      { username: 'Bob', email: 'bob@mail.com', password: '123' }
+    end
+    let(:invalid_params) { { age: 35 } }
+    context 'with valid params' do
+      let(:user_params) { valid_params }
```
Comment (qr8r) [inline spec/controllers/users_controller_spec.rb]: Why not define the valid params directly here instead

File: spec/controllers/users_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      let(:created_user) { User.last }
+      it 'creates a new User in the database' do
+        expect { post_create }.to change(User, :count).by(1)
+      end
+      it { is_expected.to redirect_to(created_user) }
+    end
+    context 'with invalid params' do
+      let(:user_params) { invalid_params }
```
Comment (qr8r) [inline spec/controllers/users_controller_spec.rb]: same as above

File: spec/controllers/users_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+    subject(:patch_update) do
+      patch :update, params: { id: user.id, user: updated_params }
+    end
+    let(:user) do
+      User.create(username: 'Bob', email: 'bob@mail.com', password: '123')
+    end
+    let(:valid_params) { { username: 'Robert' } }
+    context 'with valid params' do
+      before { log_in(user) }
+      let(:updated_params) { valid_params }
```
Comment (qr8r) [inline spec/controllers/users_controller_spec.rb]: Same same

**Review by qr8r — CHANGES_REQUESTED** [review]

File: spec/controllers/users_controller_spec.rb (hunk 3 above)
Comment (ghost) [reply spec/controllers/users_controller_spec.rb]: Thanks once again, hopefully fixed it this time.

**Review by qr8r — APPROVED** [review]
